include/fiction/utils/layout_utils.hpp

```diff
@@ -397,6 +397,84 @@ CoordinateType random_coordinate(CoordinateType coordinate1, CoordinateType coor
         return {dist_x(generator), dist_y(generator), dist_z(generator)};
     }
 }
+/**
+ * Generates a vector of all SiQAD cells within an area spanned by two SiQAD coordinates.
+ *
+ * This function calculates and returns a vector of all SiQAD cells that span the area
+ * between the northwest (cell_nw) and southeast (cell_se) cells, inclusive.
+ * The cells are generated in a top-down, left-to-right fashion within the specified area.
+ *
+ * @param cell_nw The northwest SiQAD cell defining the starting point of the area.
+ * @param cell_se The southeast SiQAD cell defining the ending point of the area.
+ * @return A vector containing all cells within the specified area.
+ */
+std::vector<siqad::coord_t> all_sidbs_in_spanned_area(const siqad::coord_t& cell_nw,
```

Comment:
Would the `foreach_coordinate` function on layouts with explicitly specified `start` and `stop` coordinates not work for your use case?

Reply:
@marcelwa Ah, I see. But does it also work for non-offset coordinates?